sgkit_vcf/vcf_partition.py

```diff
@@ -0,0 +1,142 @@
+from pathlib import Path
+from typing import Any, Optional, Sequence
+
+import numpy as np
+from cyvcf2 import VCF
+
+from sgkit.typing import PathType
+from sgkit_vcf.csi import read_csi
+from sgkit_vcf.tbi import read_tabix
+from sgkit_vcf.utils import ceildiv, get_file_length
+
+
+def region_string(contig: str, start: int, end: Optional[int] = None) -> str:
+    if end is not None:
+        return f"{contig}:{start}-{end}"
+    else:
+        return f"{contig}:{start}-"
+
+
+def get_tabix_path(vcf_path: PathType) -> Optional[Path]:
+    tbi_path = Path(vcf_path).parent / (Path(vcf_path).name + ".tbi")
+    if tbi_path.exists():
+        return tbi_path
+    else:
+        return None
+
+
+def get_csi_path(vcf_path: PathType) -> Optional[Path]:
+    csi_path = Path(vcf_path).parent / (Path(vcf_path).name + ".csi")
+    if csi_path.exists():
+        return csi_path
+    else:
+        return None
+
+
+def read_index(index_path: Path) -> Any:
+    if index_path.suffix == ".tbi":
+        return read_tabix(index_path)
+    elif index_path.suffix == ".csi":
+        return read_csi(index_path)
+    else:
+        raise ValueError("Only .tbi or .csi indexes are supported.")
+
+
+def get_sequence_names(vcf_path: Path, index: Any) -> Any:
+    try:
+        return index.sequence_names
+    except AttributeError:
+        return VCF(vcf_path).seqnames
+
+
+def partition_into_regions(
+    vcf_path: PathType,
+    *,
+    index_path: Optional[PathType] = None,
+    num_parts: Optional[int] = None,
+    target_part_size: Optional[int] = None,
+) -> Optional[Sequence[str]]:
+    """
+    Calculate genomic region strings to partition a compressed VCF or BCF file into roughly equal parts.
+
+    A .tbi or .csi file is used to find BGZF boundaries in the compressed VCF file, which are then
+    used to divide the file into parts.
+
+    The number of parts can specified directly by providing `num_parts`, or by specifying the
+    desired size of each part by providing `target_part_size`. Exactly one of `num_parts` or
```

Comment:
what are the units for ``target_part_size``  - numbers of variants? We could also imagine doing this in bytes, which might actually be more useful.

Reply:
Ah, I see it is in bytes - great! We should document this though.

Reply:
Presumably that's of the compressed file too, right?

Reply:
Correct. Added doc.

https://github.com/pystatgen/sgkit-vcf/pull/1/commits/f11198ad8525ac774781ee03c94e06093cf66fbd